composition-js/src/validate.ts

```diff
@@ -720,13 +730,29 @@ class ValidationTraversal {
       subgraphNameToGraphEnumValue,
     );
   }
+  
+  pushStack(state: ValidationState) {
```

Comment:
There's going to be a merge conflict with https://github.com/apollographql/federation/pull/3258 (the gist is "stack" gets renamed to "queue", and the `push()`/`pop()` methods renamed to `add()`/`poll()`).

Reply:
I'm going to let this get fixed in the other PR if / when we decide to merge it.